### PR TITLE
feat(donate): add donation form

### PR DESCRIPTION
**Pull request by ami**
[̲̅$̲̅(̲̅ ͡° ͜ʖ ͡°̲̅)̲̅$̲̅]

- [x] adds donation form base
- [x] connect donation form to API
- [x] add success state on donate page
- [x] make success state look good
- [x] add flairs to profile/dex pages
- [x] disable donate button when loading
- [x] adds links to donation form
- [x] make sure the email receipt looks good

fixes https://github.com/pokedextracker/pokedextracker.com/issues/83

### CODE REVIEW

**Comment by robinjoseph08** [comment]
added a few more things, but theres a few more things left to do.

you can run this and try adding things if you want, but youll have to run the `add-donations` branch of the api (and run the frontend with `NODE_ENV=local yarn start`) to actually do things. i dont think youve done that before, but i think the readme (on that branch) has everything needed.

or you can just wait until ive finished writing the tests for that branch and deploy it.

**Comment by ami** [comment]
i tried running it but i get
```
Unhandled rejection error: select "pokemon".* from "pokemon" order by "id" asc - relation "pokemon" does not exist
```
everything still runs (except i can't login and stuff) but the donation page loads but then when i submit the form it says `Failed to fetch`...

i think i shall just wait :3

**Comment by robinjoseph08** [comment]
just deployed the endpoint to staging and prod and confirmed that it worked in staging with this branch so you should be able to continue working on it! 👍

File: app/components/donate-form.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    if (session) {
+      successNote = (
+        <p>We'll be adding your profile to our project's <a href="https://github.com/pokedextracker/pokedextracker.com" target="_blank" rel="noopener noreferrer" className="link">list of supporters</a> shortly (if you're not already there). P.S.&mdash;check out your <Link className="link" to={`/u/${session.username}`}>new profile flair</Link>!</p>
+      );
+    }
+
+    if (success) {
+      header = (
+        <h1>Thanks for the donation! 💕</h1>
```
Comment (ami) [inline app/components/donate-form.jsx]: didn't know what the best way to do an emoji was... i just c/p-ed the unicode character. lmk if there's a better way...

File: app/components/donated-flair.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export function DonatedFlairComponent ({ donated }) {
+  if (!donated) {
+    return null;
+  }
+
+  return (
+    <span className="donated-flair" title="This user has donated!">🌟</span>
```
Comment (ami) [inline app/components/donated-flair.jsx]: same here

File: app/components/donate-form.jsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  render () {
+    const { session } = this.props;
+    const { amount, error, loading, success } = this.state;
+
+    let header = null;
+    let successNote = null;
+
+    if (session) {
+      successNote = (
+        <p>We'll be adding your profile to our project's <a href="https://github.com/pokedextracker/pokedextracker.com" target="_blank" rel="noopener noreferrer" className="link">list of supporters</a> shortly (if you're not already there). P.S.&mdash;check out your <Link className="link" to={`/u/${session.username}`}>new profile flair</Link>!</p>
```
Comment (ami) [inline app/components/donate-form.jsx]: i was thinking we could add a supporters section to our readme

**Comment by ami** [comment]
@robinjoseph08 updated + ready for review! lmk what you think about all copy/placement i ended up going w/ for the form success state + for the profile/dex flair!

File: app/components/donate-form.jsx (hunk 3 above)
Comment (robinjoseph08) [reply app/components/donate-form.jsx]: ya we can do that! if it gets hard to maintain, well rethink, but i highly doubt thatll happen lol. should we create an empty shell of a file (`CONTRIBUTORS.md` or something) and link to it directly? should we also mention it in the section where we say thatll well add a flair before they donate? so all the perks are mentioned upfront?

**Comment by ami** [comment]
@robinjoseph08 do you want to review this again?

**Pull request merged**